source/library/CabalGild/Action/EvaluatePragmas.hs

```diff
@@ -70,13 +70,26 @@ stripAnyExtension es p =
 
 -- | A map from field names to the set of extensions that should be discovered
 -- for that field.
+--
+-- <https://cabal.readthedocs.io/en/3.10/cabal-package.html#modules-and-preprocessors>
 extensions :: Map.Map Fields.FieldName (Set.Set String)
 extensions =
   let (=:) :: String -> [String] -> (Fields.FieldName, Set.Set String)
       k =: v = (String.toUtf8 k, Set.fromList v)
+      hs =
+        [ "chs",
+          "cpphs",
+          "gc",
+          "hs",
+          "hsc",
+          "lhs",
+          "ly",
+          "x",
+          "y"
+        ]
    in Map.fromList
-        [ "exposed-modules" =: ["hs", "lhs"],
-          "other-modules" =: ["hs", "lhs"]
+        [ "exposed-modules" =: hs,
```

Comment:
Does it still make sense to separate `exposed-modules` and `other-modules` like this? I can't think of a scenario in which the behaviour will be different w.r.t. supported file extensions.

Reply:
They should always be treated the same. That's why I pulled out the `hs` binding. 

I was trying to set this pragma up to support `signatures` (and potentially other fields) as well. However after playing around with Cabal a bit, I've discovered that it will always look for all the extensions it knows about. So for example if you have `exposed-modules: M` and `M.hsig`, it will attempt (and fail) to build the signature as a module. Similarly you could have `signatures: S` and `S.hs` — Cabal will try to build that and fail. 

So I think the right way to do this is to have a set of extensions used for discovery, and use that set for every field: `exposed-modules`, `other-modules`, and `signatures`. (Although obviously Gild doesn't support `signatures` yet. I was planning on adding it soon.)